### DISCUSSION TITLE
Permissions: Navigable O2M relationship, without being able to edit the relationship

### DISCUSSION THREAD

**Discussion by calsch03** [directus/directus#5803] [Archive]
Hallo, I've a question about one-to-many permissions.

I want to implement the following use case in Directus: I have defined a one-to-many relationship from the collection "training programs" to the collection "exercises". I've a lot of database items and I want users to be able to change the exercise from the "training programs" collection.

My problem is that you can only click on the exercise in the program collection if the user has edit permissions for the field. But I don't want to give this user role the ability to add and remove items. They only need to edit some fields on both collections.

So I have the following question: Is there a way inside Directus to only give view permissions for O2M relationships, but allow users to click on the items and edit the item in a pop-up window?

Example with only view permission:
![image](https://user-images.githubusercontent.com/19907926/119117144-3c1ea980-ba29-11eb-9fdc-f31dd65cd83c.png)

**Comment by rijkvanzanten** [comment]
Interesting!

> Is there a way inside Directus to only give view permissions for O2M relationships, but allow users to click on the items and edit the item in a pop-up window?

Not right now.. The edits made to the related item are done _through_ the current Training Exercises field, so you'll have to have edit permissions to both. We have an open issue to implement more specific o2m validation rules that should allow for this use case in the future 🙂  https://github.com/directus/directus/issues/2959
